src/charm.py

```diff
@@ -349,6 +395,33 @@ def _on_peer_relation_changed(self, event: RelationChangedEvent) -> None:
             self.unit.status = ActiveStatus()
             logger.debug(f"Instance {instance_label} is cluster member")
 
+    def _on_peer_relation_departed(self, event: RelationDepartedEvent) -> None:
```

Comment:
Remind me here, but do we need to update ipAllowList on every node?

Reply:
We're only setting it on the leader unit, but I am not sure if it needs to be set on every node. Will get Mykola's direction and make any changes in a followup. For now, the ip allow list is only updated on the leader node